ukat/vessels/phase_contrast.py

```diff
@@ -0,0 +1,288 @@
+            # Extract number pixels, area, velocity stats (min, mean, max, std)
+            # and renal blood flow (RBF)
+            self.num_pixels = np.count_nonzero(~np.isnan(self.velocity_array),
+                                               axis=(0, 1))
+            # area = (num_pixels * mm * mm * 0.01) = cm2
+            self.area = (self.num_pixels * self.pixel_spacing[0] *
+                         self.pixel_spacing[1] * 0.01)
+            self.min_velocity = np.nanmin(self.velocity_array, axis=(0, 1))
+            self.mean_velocity = np.nanmean(self.velocity_array, axis=(0, 1))
+            self.max_velocity = np.nanmax(self.velocity_array, axis=(0, 1))
+            self.std_velocity = np.nanstd(self.velocity_array, axis=(0, 1))
+            # q = (60s * cm2 * cm/s) = cm3/min = ml/min
+            self.rbf = 60 * np.array(self.area * self.mean_velocity)
+            # Mean velocity global and mean flow
+            self.mean_velocity_global = np.mean(self.mean_velocity)
+            self.mean_rbf = np.mean(self.rbf)
+            # Restrictive Index
+            mean_velocity_systole = np.max(self.mean_velocity)
+            mean_velocity_diastole = np.min(self.mean_velocity)
+            self.resistive_index = ((mean_velocity_systole -
+                                     mean_velocity_diastole) /
+                                    mean_velocity_systole)
+            # Convert any nan values to 0
+            self.velocity_array = np.nan_to_num(self.velocity_array)
+            self.mask = np.nan_to_num(self.mask)
+        else:
+            raise ValueError('The input velocity_array should be 3D.')
+
+    def get_stats_table(self):
+        """
+        Save most of PhaseContrast class attributes into a csv file.
+
+        Returns
+        ----------
+        table : pandas.DataFrame
+            Returns a table with the results/stats of each output per phase.
+        """
+        stats = {"RBF (ml/min)": self.rbf,
+                 "Area (cm2)": self.area,
+                 "Nr Pixels": self.num_pixels,
+                 "Mean Vel (cm/s)": self.mean_velocity,
+                 "Min Vel (cm/s)": self.min_velocity,
+                 "Max Vel (cm/s)": self.max_velocity,
+                 "StdDev Vel (cm/s)": self.std_velocity}
+        table = pd.DataFrame(data=stats)
+        return table
+
+    def print_stats_table(self):
+        """
+        Prints the table with the stats for each output per phase.
+        """
+        stats_table = self.get_stats_table()
+        print(tabulate(stats_table, headers='keys', tablefmt='github',
+              floatfmt='.3f'))
+
+    def save_stats_table_to_csv(self, path):
+        """
+        Save most of PhaseContrast class attributes into a csv file.
+
+        Parameters
+        ----------
+        path : str
+            Path to the desired csv file.
+        """
+        stats_table = self.get_stats_table()
+        stats_table.to_csv(path)
+
+    def plot(self, stat='default'):
```

Comment:
This looks good! The only thing I might add is the ability to save the plots e.g.
```python
def plot(self, stat='default', fname=None):
    # in all the if statements
        fig, ax = plt.subplots()
    # outside the if statements
    if fname is not None:
        fig.savefig(fname)
```